roles/spawn_container/tasks/main.yml

```diff
@@ -0,0 +1,83 @@
+---
+
+- name: Check if mount points exist
+  include_tasks: volume_check.yml
+  with_items:
+    - "{{ volumes }}"
+  when: volumes is defined and (volumes|length>0)
```

Comment:
You can eliminate all of these `volumes is defined and ...` clauses by setting an empty default for volumes and just testing the length